tests/network-tests/src/fixtures/proposalsModule.ts

```diff
@@ -732,10 +732,22 @@ export class VoteForProposalFixture extends BaseFixture {
     const councilAccounts = await this.api.getCouncilAccounts()
     this.api.treasuryTransferBalanceToAccounts(councilAccounts, proposalVoteFee)
 
-    // Approving the proposal
+    // Catch the proposal execution result
     const proposalExecutionResult = await this.api.subscribeToProposalExecutionResult(this.proposalNumber)
-    const approvals = await this.api.batchApproveProposal(this.proposalNumber)
-    approvals.map((result) => this.expectDispatchSuccess(result, 'Proposal Approval Vote Expected To Be Successful'))
+
+    // Approve proposal
+    // const approvals =
+    await this.api.batchApproveProposal(this.proposalNumber)
+
+    // Depending on the "approval_threshold_percentage" paramater for the proposal type, it might happen
+    // that the proposal can pass before all votes are cast, and this can result in an approval vote
+    // failing with a 'ProposalFinalized' Dispatch error. Important to note that depending on how many
+    // votes make it into a block, it may not always manifest becasuse the check for proposal finalization
+    // happens in on_finalize. So we no longer assert that all vote dispatches are successful.
+    // "Not taking this into account has resulted in tests failing undeterminisctly in the past.."
+    // approvals.map((result) => this.expectDispatchSuccess(result, 'Proposal Approval Vote Expected To Be Successful'))
```

Comment:
Nicely spotted!
Another way to fix this issue would be to calculate how many votes are required depending on approval quorum and threshold ([this is how I went about this on `olympia` branch](https://github.com/Joystream/joystream/blob/olympia_dev/tests/integration-tests/src/fixtures/proposals/DecideOnProposalStatusFixture.ts#L66)) in order to avoid sending any unnecessary transactions, but it's not strictly necessary.

Reply:
That's an excellent suggestion, and with your approach we would be doing extra coverage on the threshold aspect of the proposal engine implementation itself.